HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h

```diff
@@ -48,4 +48,47 @@ inline bool cudaCheck_(const char* file, int line, const char* cmd, cudaError_t
 
 #define cudaCheck(ARG) (cudaCheck_(__FILE__, __LINE__, #ARG, (ARG)))
 
+
+namespace {
+
+  template <typename T>
+  [[noreturn]] inline void abortOnCudaErrorVerbose(
+      const char* file, int line, const char* cmd, const char* error, const char* message, T const& description) {
+    std::ostringstream out;
+    out << "\n";
+    out << file << ", line " << line << ":\n";
+    out << "cudaCheck(" << cmd << ");\n";
+    out << error << ": " << message << "\n";
+    out << description << "\n";
+    throw std::runtime_error(out.str());
+  }
+
+}  // namespace
```

Comment:
Sorry for not asking earlier, why is this (and `abortOnCudaError()`) in an anonymous namespace?

Reply:
It was introduced in #333 , I don't know any more...

Reply:
How about changing it to e.g. `cudautils` then? Core guidelines [advice against anonymous namespace in headers ](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf21-dont-use-an-unnamed-anonymous-namespace-in-a-header).

Reply:
Moved everything (except the macro, of course) to `namespace cudautils`.